tweets/2022-03-31/2022-03-31-tweets.md

```diff
@@ -0,0 +1,317 @@
+Rocket News 🎉
+
+We now have support for Open Graph Images out of the box 💪
+Let's dive in and see what we can do with it!
+
+https://next.rocket.modern-web.dev/docs/guides/social-media/
+
+---
+
+First up - what the hell is an Open Graph Image?
+
+It is the preview image you see when sharing websites on Facebook, Twitter, Discord, Slack, WhatsApp, Telegram and other social media.
+As it is important that links to your website stand out from the rest of the content on social media you should add one.
+
+![Open Graph images on Twitter, Slack, Discord, Telegram, Facebook, Linkedin](./wall.png)
+
+---
+
+There are many size variations but it's usually similar.
+
+➡️ Facebook: 1200 x 628
+➡️ Twitter: 1200 x 628
+➡️ LinkedIn: 1200 x 627
+➡️ ...
+
+We will multiply that value by two to have an optimal display on retina screens.
+
+Once we have an image we add it as a meta tag to our HTML.
+
+---
+
+You may thinks "what is this fuss about?" create this image once and you are done.
+
+Not so fast! Every page should have a unique image and it should reflect the content of the page.
+
+We wanted to stand out on social media so we will need to go the extra mile.
+
+---
+
+Luckily with Rocket it's straightforward to create a unique images for each page.
+
+How does it work?
+
+1. While rendering pages with an `openGraphLayout` it will create an additional `index.opengraph.html` file
+2. During the build phase it will open that file via puppeteer and take a screenshot
+3. It injects meta tags including an absolute URL to this image
+
+---
+
+Now let's setup an Open Graph image for our site.
+
+1. Add `export const openGraphLayout = data => ...` to `recursive.data.js` so it gets injected into every page
+2. On the page itself we export some data that gets passed to the layout
+3. Define the `absoluteBaseUrl` in our rocket config as it needs to be an absolute url
+
+👉 `index.rocket.js`
+
+```js
```

Comment:
Maybe swap the order of this code block and the next one, as they are referenced in the opposite order above